### ISSUE TITLE
New install fails to boot [bare metal]

### ISSUE THREAD

**Issue by ticktockhouse** [rancher/k3os#376]
**Version (k3OS / kernel)**

k3os_version: v0.9.0
Kernel: 5.0.0-37-generic

**Architecture**
x86_64

**Describe the bug**
On bare metal:
- `sudo k3os_install` - this appears to complete successfully
- On first boot, runs resize2fs, then fails with:
```
mount: /run/k3os/target: can't find LABEL="K3OS_STATE"
```
On subsequent reboots
```[FATAL] Failed to determine boot mode```

**To Reproduce**
On bare metal
- boot from live USB
- run `sudo k3os install`, go with defaults
- reboot

**Expected behavior**
OS should boot as specified in the documentation

**Actual behavior**
See description above, OS fails to boot

**Additional context**
This is on bare metal: Intel i5, 16GB RAM 1TB SSD. I have tried booting with kernel specified as here: https://github.com/rancher/k3os/issues/60

**Comment by verbunk** [comment]
I don't notice the resizefs but I'm getting the same [FATAL] Failed to determine boot mode after an install on bare metal. It works on my test ESXi server (but that's not my deployment target :| )

k3os.mode has live and install as values, is there another one for 'installed'? I had to edit the kernel line to support console output for my system and at the same time I've run the search.fs_label K3OS_STATE and it does find the drive ~ (hd0,msdos1). The emergency terminal also has to filesystem etc.

**Comment by stuartpb** [comment]
I've tagged in a couple other issues (#458 and #462) that mention this "Failed to determine boot mode" message.

Looking in the codebase, I've tracked that failure to [here](https://github.com/rancher/k3os/blob/f77d6c6a0a31d340fcbe299709e47bda28788a15/overlay/libexec/k3os/mode#L36). Sure enough, on my system, if you look at the console output as this failure emerges, this point in the script is reached *just shortly before* the USB drive with the block label is recognized by the kernel. (Indeed, trying the `blkid -L K3OS_STATE` a second later will produce the desired output.)

Similarly, it is this failure of `blkid -L K3OS_STATE` to resolve that will break the next steps if you try to bypass this issue by merely setting `k3os.mode=disk` in the kernel command line.

The way systems like [Archiso](https://wiki.archlinux.org/index.php/archiso) handle this wait for partitions to intialize is to wait 30 seconds for the desired label to appear before interacting: I believe a similar approach would work here. I'm working on putting together a pull request for that fix, but first I'll need to see if I can get it working on my own system (having to repack the squashfs image back on to the `k3os` binary has been a significant hurdle).
